File: tusow/src/main/java/it/unibo/coordination/tusow/exceptions/ConflictError.java
package it.unibo.coordination.tusow.exceptions;

public class ConflictError extends ClientError {

    private static final int CODE = 409;
    private static final String MSG = "Conflict";

    public ConflictError() {
        super(CODE, MSG);
    }

    public ConflictError(String message) {
        super(CODE, MSG + ": " + message);
    }

    public ConflictError(String message, Throwable cause) {
        super(CODE, MSG + ": " +  message, cause);
    }

    public ConflictError(Throwable cause) {
        super(CODE, MSG, cause);
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/exceptions/ConflictError.kt
package it.unibo.coordination.tusow.exceptions

class ConflictError : ClientError {
    constructor() : super(CODE, MSG)
    constructor(message: String) : super(CODE, "$MSG: $message")
    constructor(message: String, cause: Throwable?) : super(CODE, "$MSG: $message", cause)
    constructor(cause: Throwable?) : super(CODE, MSG, cause)

    companion object {
        private const val CODE = 409
        private const val MSG = "Conflict"
    }
}